### PR TITLE
Support leading and trailing spaces in query options

### PR DESCRIPTION
**Pull request by kosinsky**
### Issues

*This pull request fixes issue #1822.*

### Description

Adds Trim() to allow following queries ` $filter=Filter` (notice leading space), `$filter =Filter` (notice space before =), `%20$filter=Filter` 

### Checklist (Uncheck if it is not completed)

- [x] *Test cases added*
- [x] *Build and test with one-click build and test script passed*

### CODE REVIEW

File: test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Query/ODataQueryOptionTest.cs [hunk 1]
```diff
@@ -264,6 +264,37 @@ public void CanExtractQueryOptionsCorrectly()
             Assert.Equal("DeltaToken", queryOptions.RawValues.DeltaToken);
         }
 
+
+        [Theory]
+        [InlineData(" $filter=Filter& $select=Select& $orderby=OrderBy& $expand=Expand& $top=10& $skip=20& $count=true& $skiptoken=SkipToken& $deltatoken=DeltaToken")]
+        [InlineData("%20$filter=Filter&%20$select=Select&%20$orderby=OrderBy&%20$expand=Expand&%20$top=10&%20$skip=20&%20$count=true&%20$skiptoken=SkipToken&%20$deltatoken=DeltaToken")]
+        [InlineData("$filter =Filter&$select =Select&$orderby =OrderBy&$expand =Expand&$top =10&$skip =20&$count =true&$skiptoken =SkipToken&$deltatoken =DeltaToken")]
+        public void CanExtractQueryOptionsWitLeadingSpacesCorrectly(string clause)
```
Comment (xuzhg) [inline test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Query/ODataQueryOptionTest.cs]: >Wit [](start = 42, length = 3)

With ?

File: test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Query/ODataQueryOptionTest.cs [hunk 2]
```diff
@@ -264,6 +264,37 @@ public void CanExtractQueryOptionsCorrectly()
             Assert.Equal("DeltaToken", queryOptions.RawValues.DeltaToken);
         }
 
+
+        [Theory]
+        [InlineData(" $filter=Filter& $select=Select& $orderby=OrderBy& $expand=Expand& $top=10& $skip=20& $count=true& $skiptoken=SkipToken& $deltatoken=DeltaToken")]
```
Comment (xuzhg) [inline test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Query/ODataQueryOptionTest.cs]: >$filter=Filter& $select=Select& $orderby=OrderBy [](start = 22, length = 48)

Maybe add the example of tailing ?

File: test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Query/ODataQueryOptionTest.cs [hunk 3]
```diff
@@ -264,6 +264,37 @@ public void CanExtractQueryOptionsCorrectly()
             Assert.Equal("DeltaToken", queryOptions.RawValues.DeltaToken);
         }
 
+
+        [Theory]
```
Comment (xuzhg) [inline test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Query/ODataQueryOptionTest.cs]: remove unused the white line?

File: test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Query/ODataQueryOptionTest.cs [hunk 4]
```diff
@@ -264,6 +264,37 @@ public void CanExtractQueryOptionsCorrectly()
...
+        [InlineData(" $filter=Filter& $select=Select& $orderby=OrderBy& $expand=Expand& $top=10& $skip=20& $count=true& $skiptoken=SkipToken& $deltatoken=DeltaToken")]
+        [InlineData("%20$filter=Filter&%20$select=Select&%20$orderby=OrderBy&%20$expand=Expand&%20$top=10&%20$skip=20&%20$count=true&%20$skiptoken=SkipToken&%20$deltatoken=DeltaToken")]
+        [InlineData("$filter =Filter&$select =Select&$orderby =OrderBy&$expand =Expand&$top =10&$skip =20&$count =true&$skiptoken =SkipToken&$deltatoken =DeltaToken")]
+        public void CanExtractQueryOptionsWitLeadingSpacesCorrectly(string clause)
+        {
+            var model = new ODataModelBuilder().Add_Customer_EntityType().Add_Customers_EntitySet().GetEdmModel();
```
Comment (xuzhg) [inline test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Query/ODataQueryOptionTest.cs]: > var model = new ODataModelBuilder() [](start = 11, length = 36)

would you help add // Arrange // Act // Assert

And help that in the above one?

File: test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Query/ODataQueryOptionTest.cs (hunk 2 above)
Comment (kosinsky) [reply test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Query/ODataQueryOptionTest.cs]: Last InlineData  (line 271)  does trailing `$filter =Filter...`

Comment (kosinsky) [reply test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Query/ODataQueryOptionTest.cs]: I added two more cases trailing %20  and spaces before and after

**Comment by kosinsky** [comment]
@xuzhg do you have any other feedback?

**Comment by xuzhg** [comment]
> @xuzhg do you have any other feedback?

So far, it looks good to me.

**Pull request merged**